### DISCUSSION TITLE
What to do next?

### DISCUSSION THREAD

**Discussion by thany** [Vinzent03/obsidian-git#786] [Q&A]
I've installed the plugin, set up my vault as a git repo, hooked it up to github, and pushed my first commit containing the whole contents of my vault.

Everything works, as far as I can tell, from the command line.

But now when I make any change to any file in my vault - what do I do? I'd like it to sync it to github by creating a commit and pushing it. That's kind of how it works. It doesn't do that automatically, so I open the side panel. But, what do I do there? I don't get it.

I am actually quite familiar with Git as source control, being a frontend developer myself. So it's not like I don't know what git is or anything. But I do expect my editor to show unstaged changes *at the very least* when I make some and save them. It doesn't do that at all. I just clicked the "commit-and-sync" button, hoping it would be a one-click-sync-all-and-don't-think-about-it kind of button.

It just looks like this:
![image](https://github.com/user-attachments/assets/56d1d370-83b0-467b-a955-181e28d782ef)

No matter what I edit in documents. And the refresh icon keeps spinning, as if it never finishes.

Also in the plugin settings, it keeps saying "Git is not ready, etc". How does it arrive at that conclusion? Afaik, git is well and truly ready, I mean didn't I just set it up and pushed my first commit?...

So the bottomline is: what do I do now? Documentation is lacking in this regard.

Edit:

Things started happening when I did a pull from the sidebar. Why did I have to do that? Once `git push` works, you don't normally have to pull before the repo is ready for use (or whatever that means).

Is there a log somewhere, where I can find information for when it breaks? Git GUIs usually have something.

**Comment by Vinzent03** [comment]
Hm that's strange. May you try restarting Obsidian and share the content of the console (`CTRL+SHIFT+I`)?

**Reply by thany** [reply]
Well, currently it only shows

```
loading Git plugin: v2.27.0
```

And that's it, right after starting Obsidian. I can't reproduce the trouble I was having anymore (and frankly I don't want to, I would have to fiddle around to get things working again, assuming I can actually reproduce the problem in the first place).

I'm just wondering if this plugin really needs to be a fully featured Git client, because maybe one giant "sync now" button is enough. It doesn't need to be as complicated as a Git client needs to be for a developer, does it? With respect, perhaps simplifying the plugin to cater more to the kind of person just writing articles and whatnot, rather than developing programs, might reveal the plugin to be slightly on the complicated side and reveal why it doesn't "just work" after setting things up.
